pkg/plugins/policies/meshtrace/plugin/v1alpha1/plugin.go

```diff
@@ -20,11 +32,160 @@ func NewPlugin() core_plugins.Plugin {
+	serviceName := dataplane.Spec.GetIdentifyingService()
+	listener := listenerResource.Resource.(*envoy_listener.Listener)
+	rawConf := rules.Rules[0].Conf
+	conf := rawConf.(*api.MeshTrace_Conf)
+	destination := ""
+
+	if listenerResource.Origin == generator.OriginOutbound {
+		i := slices.IndexFunc(dataplane.Spec.Networking.Outbound, func(outbound *mesh_proto.Dataplane_Networking_Outbound) bool {
+			outboundInterface := dataplane.Spec.Networking.ToOutboundInterface(outbound)
+			address := listener.GetAddress().GetSocketAddress()
+			return outboundInterface.DataplaneIP == address.GetAddress() && outboundInterface.DataplanePort == address.GetPortValue()
+		})
+
+		if i != -1 {
+			outbound := dataplane.Spec.Networking.GetOutbound()[i]
+			destination = outbound.GetTagsIncludingLegacy()[mesh_proto.ServiceTag]
+		}
+	}
+
+	configurer := plugin_xds.Configurer{
+		Conf:             conf,
+		Service:          serviceName,
+		TrafficDirection: listener.TrafficDirection,
+		Destination:      destination,
+	}
+
+	for _, chain := range listener.FilterChains {
+		if err := configurer.Configure(chain); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func applyToClusters(rules xds.SingleItemRules, rs *xds.ResourceSet, proxy *xds.Proxy) error {
+	rawConf := rules.Rules[0].Conf
+
+	conf := rawConf.(*api.MeshTrace_Conf)
+
+	backend := conf.GetBackends()[0]
+	if backend == nil {
+		return nil
+	}
+
+	var endpoint *xds.Endpoint
+	var err error
+	var provider string
+
+	if backend.GetZipkin() != nil {
+		endpoint, err = endpointForZipkin(backend.GetZipkin())
+		provider = "zipkin"
+		if err != nil {
+			return errors.Wrap(err, "could not generate zipkin cluster")
+		}
+	} else {
+		endpoint, err = endpointForDatadog(backend.GetDatadog())
+		provider = "datadog"
+		if err != nil {
+			return errors.Wrap(err, "could not generate zipkin cluster")
+		}
+	}
+
+	res, err := clusters.NewClusterBuilder(proxy.APIVersion).
+		Configure(clusters.ProvidedEndpointCluster(plugin_xds.GetTracingClusterName(provider), proxy.Dataplane.IsIPv6(), *endpoint)).
+		Configure(clusters.ClientSideTLS([]xds.Endpoint{*endpoint})).
+		Configure(clusters.DefaultTimeout()).
+		Build()
+	if err != nil {
+		return err
+	}
+
+	rs.Add(&xds.Resource{Name: plugin_xds.GetTracingClusterName(provider), Origin: MeshTraceOrigin, Resource: res})
+
+	return nil
+}
+
+func endpointForZipkin(cfg *api.MeshTrace_ZipkinBackend) (*xds.Endpoint, error) {
+	url, err := net_url.ParseRequestURI(cfg.Url)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid URL of Zipkin")
+	}
+	port, err := strconv.Atoi(url.Port())
+	if err != nil {
+		return nil, err
+	}
+	return &xds.Endpoint{
+		Target: url.Hostname(),
+		Port:   uint32(port),
+		ExternalService: &xds.ExternalService{
+			TLSEnabled:         url.Scheme == "https",
+			AllowRenegotiation: true,
+		},
+	}, nil
+}
+
+func endpointForDatadog(cfg *api.MeshTrace_DatadogBackend) (*xds.Endpoint, error) {
+	if cfg.Port > 0xFFFF || cfg.Port < 1 {
```

Comment:
We check this on validation but it was in the old policy so I left it out, should I remove this or is there other way to get here bypassing validation?

Reply:
> should we make this coherent between the 2?

They are different between the two of them in Envoy config. We can't go for `host/port` because then we don't have paths in `host`. We could use `url` for both but anything other than `host/port` will be useless for datadog - we could add validation to not allow that. WDYT?

Reply:
Changed, let me know what you think :)